conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,642 @@
+        """
+        dirs = self._get_component_dirs()
+        flags = self._get_component_flags()
+        comp_content_gen = DepComponentContentGenerator(self._dep, self._name, dirs, flags)
+        comp_content = comp_content_gen.content()
+        return comp_content
+
+
+class DepGenerator:
+    """
+    Process a dependency cpp_info variables and generate its Makefile content
+    """
+
+    def __init__(self, dependency: object):
+        self._dep = dependency
+        self._info = MakeInfo(self._dep.ref.name, [], [])
+
+    def make_global_info(self) -> MakeInfo:
+        """
+        Remove duplicated folders and flags from current MakeInfo
+        :return: Dependency folder and flags
+        """
+        self._info.dirs = list(set(self._info.dirs))
+        self._info.flags = list(set(self._info.flags))
+        return self._info
+
+    def _get_dependency_dirs(self, root: str, dependency: object) -> dict:
+        """
+        List regular directories from cpp_info and format them to be used in the makefile
+        :param root: Package root folder
+        :param dep: Dependency object
+        :return: A dictionary with regular folder name and its formatted path
+        """
+        dirs = {}
+        for var, prefix in _common_cppinfo_dirs().items():
+            cppinfo_value = getattr(dependency.cpp_info, var)
+            formatted_dirs = _get_formatted_dirs(cppinfo_value, root, _makefy(dependency.ref.name))
+            if formatted_dirs:
+                self._info.dirs.append(var)
+                var = var.replace("dirs", "_dirs")
+                dirs[var] = [_conan_prefix_flag(prefix) + it for it in formatted_dirs]
+        return dirs
+
+    def _get_dependency_flags(self, dependency: object) -> dict:
+        """
+        List common variables from cpp_info and format them to be used in the makefile
+        :param dep: Dependency object
+        """
+        flags = {}
+        for var, prefix_var in _common_cppinfo_variables().items():
+            cppinfo_value = getattr(dependency.cpp_info, var)
+            # Use component cpp_info info when does not provide any value
+            if not cppinfo_value and hasattr(dependency.cpp_info, "components"):
+                cppinfo_value = [f"$(CONAN_{var.upper()}_{_makefy(dependency.ref.name)}_{_makefy(name)})" for name, obj in dependency.cpp_info.components.items() if getattr(obj, var.lower())]
+                # avoid repeating same prefix twice
+                prefix_var = ""
+            if "flags" in var:
+                cppinfo_value = [var.replace('"', '\\"') for var in cppinfo_value]
+            if cppinfo_value:
+                self._info.flags.append(var)
+                flags[var] = [_conan_prefix_flag(prefix_var) + it for it in cppinfo_value]
+        return flags
+
+    def _get_sysroot(self, root: str) -> list:
+        """
+        Get the sysroot of the dependency. Sysroot is a list of directories, or a single directory
```

Comment:
So far we haven't treated ``sysroot`` as multiple directories, maybe this would be better to align with others and treat it jointly with other generators behavior.

Reply:
Indeed is not usual having multiple sysroot, but only one when cross-building. TBH, it seems be a bug, because returning `['']` seems be something mixed, it should be empty string, but then was converted to a list to keep same pattern as other cpp_info dirs